Dockerfile

```diff
@@ -4,30 +4,24 @@
 # NOTE: this is not provided for production usage.
 FROM mozillamarketplace/centos-mysql-mkt:0.2
 
-RUN yum install -y supervisor
-RUN yum install -y bash-completion
-RUN yum install -y cronie
+RUN yum install -y supervisor bash-completion cronie && yum clean all
 
 ENV IS_DOCKER 1
 
-RUN mkdir -p /pip/{cache,build}
-ADD requirements /pip/requirements
-WORKDIR /pip
+# Ship the source in the container, its up to docker-compose to override it
+# if it wants to.
+COPY . /srv/solitude
+RUN cd /srv/solitude && git show -s --pretty="format:%h" > git-rev.txt
+
 # Download this securely from pyprepo first.
 RUN pip install --no-deps --find-links https://pyrepo.addons.mozilla.org/ peep
 RUN peep install \
-    --build /pip/build \
-    --download-cache /pip/cache \
+    --download-cache /tmp/pip/cache \
```

Comment:
No pip creates the directory if it doesn't exist. If peep doesn't use it, its gone... wasn't quite sure of the point in this context anyway.